boxes/generators/stackablebin.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (C) 2013-2016 Florian Festi
+#
+#   This program is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   This program is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU General Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License
+#   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from boxes import *
+
+
+class StackableBinEdge(edges.BaseEdge):
+    char = "B"
+
+    def __call__(self, length, **kw):
+        f = self.settings.front
+        a1 = math.degrees(math.atan(f/(1-f)))
+        a2 = 45 + a1
+        self.corner(-a1)
+
+        self.edges["e"](self.settings.y* (f**2+(1-f)**2)**0.5)
+        self.corner(a2)
+        self.edges["f"](self.settings.y*f*2**0.5)
+
+        self.corner(-45)
+
+    def margin(self) -> float:
+        return self.settings.y * self.settings.front
+
+class StackableBinSideEdge(StackableBinEdge):
+    char = 'b'
+
+class StackableBin(Boxes):
+    """Stackable bin base on bintray"""
+
+    ui_group = "Shelf"
+
+    def __init__(self) -> None:
+        Boxes.__init__(self)
+        self.buildArgParser("outside")
+        self.buildArgParser(x=70, y=50, h=100)
+        self.addSettingsArgs(edges.FingerJointSettings, surroundingspaces=0.5)
+        self.argparser.add_argument(
+            "--front", action="store", type=float, default=0.4,
+            help="fraction of bin height covered with slope")
+
+    def render(self):
+        if self.outside:
+            self.x = self.adjustSize(self.x)
+            self.y = self.adjustSize(self.y)
+            self.h = self.adjustSize(self.h, e2=False)
```

Comment:
This should be adjustsize(self.h, "s", "S") to adjust for the stackable edges

Reply:
done